recipes-domu/domu-image-android/domu-image-android.bb

```diff
@@ -1,76 +1,122 @@
-SUMMARY = "Base for Android images"
+SUMMARY = "Base for build of Android"
 
 LICENSE = "MIT"
 
-inherit build_yocto
-inherit xt_quirks
-inherit xt_reconstruct
+require inc/xt_shared_env.inc
 
-FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
-FILESEXTRAPATHS_prepend := "${THISDIR}/../../classes:"
+# Important note.
+# We use REPODIR and SRC_URI with 'directpath=1' to bypass fetcher's
+# modification of download path. This is done for performance reasons.
+#
+# 'Out-of-stock' repo fetcher gets sources into ${DL_DIR}/repo/xxx folder,
+# then tar/gzip them and then ungzip to ${S}.
+# These steps add almost one hour to fetching on PC.
+#
+# With modifications to xt-distro's repo fetcher (adding 'depth' and 'directpath'
+# parameters to SRC_URI) and using REPODIR instead of DL_DIR (to avoid copying),
+# we have do_fetch() completed within 20 minutes.
+#
+# Some details:
+# 'depth=1' reduces amount of data downloaded from server and is most significant
+#           reducer of fetch time
+# 'REPODIR' is used to download sources directly into required place,
+#           and to avoid copying of tens of gigabytes from ${DL_DIR}
+# 'directpath=1' is used to suppress creation of subfolders to ${REPODIR}.
+#           This is required for correct work of do_compile() with downloaded sources.
 
-S = "${WORKDIR}/repo"
+REPODIR = "${SSTATE_DIR}/../android_sources"
+S = "${REPODIR}"
 
-################################################################################
-# set to AUTOREV so we can override this while reconstructing this build with
-# specific commit ID
-################################################################################
-SRCREV = "${AUTOREV}"
+# depth=1 reduce fetch time in half: ~(10-14)min instead of 23min
+SRC_URI = "repo://github.com/xen-troops/android_manifest;protocol=https;branch=android-11-master;manifest=doma.xml;depth=1;directpath=1"
+# If XT_ANDROID_PREBUILDS_DIR is not set, we need to use proprietary sources for build of graphics
+python __anonymous () {
```

Comment:
1. there 3 instances of the anonymous
2. this approach does not work if src_uri will be modified  - groups=all will be added to the wrong place
'3. why required 'directpath=1; is the part of the url but 'groups=all'  not ?
4. if some attributes are required (like groups and directpath), it is reasonable to check their availability during the recipe cinstruction, and stop processing if the attributes are absent

Reply:
1. I consider anonymous as logically different and located close to variables they modify, but I will unite them and restructure sources. WILL REWORK
2. For now (at this stage of modifications of android's build) SRC_USI for domu-image-android is not intended to contain anything except doma.xml. This is the only purpose of this receipt - to lunch android's build. But I will rename SRC_URI and provide corresponding comments. WILL REWORK
3. groups is not required part of URL. It is appended only if we build proprietary sources. In this case we need to provide 'groups=all' so repo will get proprietary sources. If we use prebuilt graphic binaries, we do not use groups.
4. I have provided special description from line 7 to line 25 explaining why new parameters are used and what they are doing. So, we can't stop processing because it need to be decided and used by developer in some cases, and in other cases these parameters are not used (like for regular manifests for dom0 and domd).